### PR TITLE
Changes for Summit.

### PR DESCRIPTION
**Pull request by corbett5**
Related to:
https://github.com/GEOSX/integratedTests/pull/127

### CODE REVIEW

File: src/coreComponents/fileIO/silo/SiloFile.cpp [hunk 1]
```diff
@@ -1314,8 +1315,8 @@ void SiloFile::writeElementRegionSilo( ElementRegionBase const & elemRegion,
         // check if the field actually exists / plotted on the current subregion
         if( viewPointers[esr].count( fieldName ) > 0 )
         {
-          Wrapper< arrayType > const & sourceWrapper = dynamicCast< Wrapper< arrayType > const & >( *(viewPointers[esr][fieldName]));
-          traits::ViewTypeConst< arrayType > const sourceArray = sourceWrapper.reference().toViewConst();
+          WrapperType const & sourceWrapper = dynamic_cast< WrapperType const & >( *(viewPointers[esr].at( fieldName )) );
```
Comment (corbett5) [inline src/coreComponents/fileIO/silo/SiloFile.cpp:1318]: This is the only code change. I was getting some really strange error, probably a compiler bug.

File: src/cmake/thirdparty/SetupGeosxThirdParty.cmake [hunk 2]
```diff
@@ -425,15 +425,16 @@ if(DEFINED HYPRE_DIR)
...
-      set(ENABLE_HYPRE ON CACHE BOOL "")
-    endif()
+    # if( ENABLE_CUDA AND ( NOT ENABLE_HYPRE_CUDA ) )
+    #   set(ENABLE_HYPRE OFF CACHE BOOL "" FORCE)
+    #   if( GEOSX_LA_INTERFACE STREQUAL "Hypre")
+    #     message( FATAL_ERROR "Hypre LAI selected, but ENABLE_HYPRE_CUDA not ON while ENABLE_CUDA is ON.")
+    #   endif()
+    # else()
+    #   set(ENABLE_HYPRE ON CACHE BOOL "")
+    # endif()
```
Comment (rrsettgast) [inline src/cmake/thirdparty/SetupGeosxThirdParty.cmake:435]: This is a little wonky, but can't we just turn hypre off?

File: src/docs/doxygen/GeosxConfig.hpp [hunk 3]
```diff
@@ -72,11 +72,11 @@
 #define GEOSX_USE_PETSC
 
 /// Choice of global linear algebra interface (CMake option GEOSX_LA_INTERFACE)
-#define GEOSX_LA_INTERFACE Hypre
+#define GEOSX_LA_INTERFACE Trilinos
```
Comment (rrsettgast) [inline src/docs/doxygen/GeosxConfig.hpp:75]: you can build this on quartz to get this file back to the default.

**Review by rrsettgast — APPROVED** [review]

File: src/docs/doxygen/GeosxConfig.hpp (hunk 3 above)
Comment (corbett5) [reply src/docs/doxygen/GeosxConfig.hpp]: Done.

**Pull request merged**